src/components/WorkQueueStatusBadge.vue

```diff
@@ -1,28 +1,36 @@
 <template>
   <template v-if="workQueue && workQueueStatus">
     <p-tag class="work-queue-status-badge" :class="classes">
-      <WorkQueueStatusIcon :work-queue-name="workQueueName" class="work-queue-status-badge__icon" />
+      <WorkPoolQueueStatusIcon v-if="can.access.workQueueStatus" :work-pool-queue="workQueue" />
+      <WorkQueueStatusIcon v-else :work-queue-name="workQueueName" class="work-queue-status-badge__icon" />
       {{ label }}
     </p-tag>
   </template>
 </template>
 
 <script lang="ts" setup>
   import { computed } from 'vue'
+  import WorkPoolQueueStatusIcon from '@/components/WorkPoolQueueStatusIcon.vue'
   import WorkQueueStatusIcon from '@/components/WorkQueueStatusIcon.vue'
-  import { useWorkQueueStatus } from '@/compositions'
+  import { useCan, useWorkQueueStatus } from '@/compositions'
   import { WorkQueue } from '@/models'
 
   const props = defineProps<{
     workQueue: WorkQueue,
   }>()
 
+  const can = useCan()
+
   const workQueueName = computed(() => props.workQueue.name)
   const workQueueId = computed(() => props.workQueue.id)
   const { workQueueStatus } = useWorkQueueStatus(workQueueId)
   const healthy = computed(() => workQueueStatus.value?.healthy)
 
   const label = computed(() => {
+    if (can.access.workQueueStatus) {
+      return { 'ready': 'Ready', 'not_ready': 'Not Ready', 'paused': 'Paused' }[props.workQueue.status]
```

Comment:
This is weird but I'm guessing it's in lieu of more formal mapping for an otherwise deprecated set of components?

Reply:
yea here's to hoping this file gets deleted before anyone looks at it again 🙈